orc8r/gateway/python/scripts/register.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import argparse
+import sys
+import textwrap
+from typing import List
+
+import snowflake
+from magma.common.cert_utils import load_public_key_to_base64der
+from magma.common.service_registry import ServiceRegistry
+from magma.configuration.service_configs import save_override_config
+from orc8r.protos.bootstrapper_pb2 import (
+    ChallengeKey,
+    RegisterRequest,
+    RegisterResponse,
+)
+from orc8r.protos.bootstrapper_pb2_grpc import RegistrationStub
+from orc8r.protos.identity_pb2 import AccessGatewayID
+
+
+def register_handler(client: RegistrationStub, args: List[str]) -> RegisterResponse:
+    """
+    Register a device and retrieves its control proxy
+    Args:
+        client: Registration stub
+        args: command line arguments
+    Returns:
+        RegisterResponse: response from gRPC call, either error or the control_proxy
+    """
+    req = RegisterRequest(
+        token=args.token,
+        hwid=AccessGatewayID(
+            id=snowflake.snowflake(),
+        ),
+        challenge_key=ChallengeKey(
+            key=load_public_key_to_base64der("/var/opt/magma/certs/gw_challenge.key"),
+        ),
+    )
+
+    res = client.Register(req)
+    if res.HasField("error"):
+        raise Exception(res.error)
+
+    msg = textwrap.dedent(
+        """
+        > Successfully registered gateway
+        Hardware ID
+        -----------
+        {}
+        Challenge Key
+        -----------
+        {}
+        Control Proxy
+        -----------
+        {}
+        """,
+    )
+    print(msg.format(req.hwid, req.challenge_key, res.control_proxy))
+
+    return res
+
+
+def main():
+    """Register a gateway"""
+    parser = argparse.ArgumentParser(description="Register a gateway.")
+    parser.add_argument(
+        "domain",
+        metavar="DOMAIN_NAME",
+        type=str,
+        help="orc8r's domain name",
+    )
+    parser.add_argument(
+        "token",
+        metavar="REGISTRATION_TOKEN",
+        type=str,
+        help="registration token after API call",
+    )
```

Comment:
Is there a way to mark these arguments as required (and fail/throw err when they're not present) ?

Reply:
Soo because these are considered `positional arguments` , python's `argparse` already throws an error when these are not present